### PR TITLE
pd client: Support getting timestamp from PD

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

## What have you changed? (mandatory)

This PR adds function `get_timestamp` to PdClient so that TiKV can get tso from PD now.

PD's TSO is a stream interface, and the current implementation is very naive: for each invoking to `get_timestamp`, it establishes a stream, get the timestamp and close it. It's ok to use it in the commit_ts optimization. If someday we need to use it more frequently, we can choose a better implementation then.

## What are the type of the changes? (mandatory)

- New feature (change which adds functionality)

## How has this PR been tested? (mandatory)

By CI

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

### CODE REVIEW

**Comment by MyonKeminta** [comment]
@hicqu PTAL

**Comment by MyonKeminta** [comment]
@zhangjinpeng1987 PTAL

File: src/pd/client.rs [hunk 1]
```diff
@@ -526,4 +528,37 @@ impl PdClient for RpcClient {
...
+        req.set_header(self.header());
+        req.set_count(1);
+
+        let executor = move |client: &RwLock<Inner>, req: pdpb::TsoRequest| {
+            let option = CallOption::default().timeout(Duration::from_secs(REQUEST_TIMEOUT));
+            let (sender, receiver) = client.rl().client.tso_opt(option).unwrap();
+            let send = sender.send((req, WriteFlags::default().buffer_hint(false)));
+            Box::new(
+                send.and_then(|mut s| {
+                    let _ = s.close();
```
Comment (overvenus) [inline src/pd/client.rs]: We should not ignore the return value.

File: src/pd/client.rs [hunk 2]
```diff
@@ -526,4 +528,37 @@ impl PdClient for RpcClient {
...
+                    receiver.collect().map(|mut v| v.remove(0))
```
Comment (overvenus) [inline src/pd/client.rs]: The `collect` allocates a `Vec`, what about `into_future`?

File: src/pd/client.rs [hunk 3]
```diff
@@ -37,6 +37,8 @@ const CLIENT_PREFIX: &str = "pd";
 const MAX_RETRY_TIMES: u64 = 10;
 const RETRY_INTERVAL_MS: u64 = 300;
 
+const TSO_PHYSICAL_SHIFT: u64 = 18;
```
Comment (overvenus) [inline src/pd/client.rs]: Could you add some comments?

File: src/pd/client.rs (hunk 1 above)
Comment (MyonKeminta) [reply src/pd/client.rs]: Can we unwrap it? Or should we just print a error log for it?

Comment (MyonKeminta) [reply src/pd/client.rs]: Can we use `cancel` here?

**Comment by MyonKeminta** [comment]
Oh, sorry. I need to add test for it.

**Comment by MyonKeminta** [comment]
The current implementation in this PR is not very elegant. Since this is not necessary if we do not continue the optimization of calculated commit ts and single region 1pc, close it and reopen it when necessary.

**Pull request closed without merging**